WalletWasabi.Fluent/ViewModels/Wallets/Home/Tiles/PrivacyRing/PrivacyBarViewModel.cs

```diff
@@ -5,67 +5,74 @@
 using System.Linq;
 using System.Reactive;
 using System.Reactive.Linq;
-using Avalonia.Threading;
+using NBitcoin;
 using WalletWasabi.Fluent.Extensions;
 using WalletWasabi.Fluent.Helpers;
 using WalletWasabi.Fluent.Models;
 using WalletWasabi.Wallets;
+using System.Reactive.Disposables;
 
 namespace WalletWasabi.Fluent.ViewModels.Wallets.Home.Tiles.PrivacyRing;
 
-public partial class PrivacyBarViewModel : ViewModelBase
+public partial class PrivacyBarViewModel : ActivatableViewModel
 {
-	private readonly SourceList<PrivacyBarItemViewModel> _itemsSourceList = new();
-	private IObservable<Unit> _coinsUpdated;
+	private readonly WalletViewModel _walletViewModel;
+	private const decimal GapBetweenSegments = 1.5m;
+	private const decimal EnlargeThreshold = 2m;
+	private const decimal EnlargeBy = 1m;
 
 	[AutoNotify] private double _width;
 
 	public PrivacyBarViewModel(WalletViewModel walletViewModel)
 	{
+		_walletViewModel = walletViewModel;
 		Wallet = walletViewModel.Wallet;
+	}
+
+	public ObservableCollectionExtended<PrivacyBarItemViewModel> Items { get; } = new();
+
+	public Wallet Wallet { get; }
 
-		_itemsSourceList
+	[System.Diagnostics.CodeAnalysis.SuppressMessage("Reliability", "CA2000:Dispose objects before losing scope", Justification = "Uses DisposeWith()")]
+	protected override void OnActivated(CompositeDisposable disposables)
+	{
+		var itemsSourceList = new SourceList<PrivacyBarItemViewModel>().DisposeWith(disposables);
+
+		itemsSourceList
 			.Connect()
 			.ObserveOn(RxApp.MainThreadScheduler)
 			.Bind(Items)
 			.DisposeMany()
-			.Subscribe();
+			.Subscribe()
+			.DisposeWith(disposables);
 
-		_coinsUpdated =
-			walletViewModel.UiTriggers.PrivacyProgressUpdateTrigger
-						  .Merge(walletViewModel
+		var coinsUpdated =
+			_walletViewModel.UiTriggers.PrivacyProgressUpdateTrigger
+						  .Merge(_walletViewModel
```

Comment:
As we discussed in your other PR, CJ trigger is not needed even on the privacy bar. I don't want to change it in my PR as it will cause conflict here. Would you remove it here?

Reply:
Done